src/rm_topk.c

```diff
@@ -332,25 +332,32 @@ static size_t TopKMemUsage(const void *value) {
 
 int TopKModule_onLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     // TODO: add option to set defaults from command line and in program
-    RedisModuleTypeMethods tm = {.version = REDISMODULE_TYPE_METHOD_VERSION,
-                                 .rdb_load = TopKRdbLoad,
-                                 .rdb_save = TopKRdbSave,
-                                 .aof_rewrite = RMUtil_DefaultAofRewrite,
-                                 .mem_usage = TopKMemUsage,
-                                 .free = TopKFree,
-                                 .defrag = TopKDefrag};
+    RedisModuleTypeMethods tm = {
+        .version = REDISMODULE_TYPE_METHOD_VERSION,
+        .rdb_load = TopKRdbLoad,
+        .rdb_save = TopKRdbSave,
+        .aof_rewrite = RMUtil_DefaultAofRewrite,
+        .mem_usage = TopKMemUsage,
+        .free = TopKFree,
+        .defrag = TopKDefrag,
+    };
 
     TopKType = RedisModule_CreateDataType(ctx, "TopK-TYPE", TOPK_ENC_VER, &tm);
-    if (TopKType == NULL)
+    if (TopKType == NULL) {
         return REDISMODULE_ERR;
+    }
 
-    RMUtil_RegisterWriteDenyOOMCmd(ctx, "topk.reserve", TopK_Create_Cmd);
-    RMUtil_RegisterWriteDenyOOMCmd(ctx, "topk.add", TopK_Add_Cmd);
-    RMUtil_RegisterWriteDenyOOMCmd(ctx, "topk.incrby", TopK_Incrby_Cmd);
-    RMUtil_RegisterReadCmd(ctx, "topk.query", TopK_Query_Cmd);
-    RMUtil_RegisterWriteCmd(ctx, "topk.count", TopK_Count_Cmd);
-    RMUtil_RegisterReadCmd(ctx, "topk.list", TopK_List_Cmd);
-    RMUtil_RegisterReadCmd(ctx, "topk.info", TopK_Info_Cmd);
+#define MODULE_ACL_CATEGORY_NAME "topk"
+    RegisterAclCategory(ctx);
+    RegisterCommandWithModesAndAcls(ctx, "topk.reserve", TopK_Create_Cmd, "write deny-oom",
+                                    "write fast");
+    RegisterCommandWithModesAndAcls(ctx, "topk.add", TopK_Add_Cmd, "write deny-oom", "write");
+    RegisterCommandWithModesAndAcls(ctx, "topk.incrby", TopK_Incrby_Cmd, "write deny-oom", "write");
+    RegisterCommandWithModesAndAcls(ctx, "topk.query", TopK_Query_Cmd, "readonly", "read");
+    RegisterCommandWithModesAndAcls(ctx, "topk.count", TopK_Count_Cmd, "write", "write");
```

Comment:
check if should be read

Reply:
you're right, should be read. `GetTopKKey(ctx, argv[1], &topk, REDISMODULE_READ)`